tests/unit/setup-rendering-context-test.js

```diff
@@ -115,4 +117,187 @@ module('setupRenderingContext', function(hooks) {
 
     assert.equal(this.element.textContent, 'outerinnerouter');
   });
+
+  test('can use the component helper in its layout', async function(assert) {
+    this.owner.register('template:components/x-foo', hbs`x-foo here`);
+
+    await this.render(hbs`{{component 'x-foo'}}`);
+
+    assert.equal(this.element.textContent, 'x-foo here');
+  });
+
+  test('can create a component instance for direct testing without a template', function(assert) {
+    this.owner.register(
+      'component:foo-bar',
+      Component.extend({
+        someMethod() {
+          return 'hello thar!';
+        },
+      })
+    );
+
+    let subject;
+    if (hasEmberVersion(2, 12)) {
+      subject = this.owner.lookup('component:foo-bar');
+    } else {
+      subject = this.owner._lookupFactory('component:foo-bar').create();
+    }
+
+    assert.equal(subject.someMethod(), 'hello thar!');
+  });
+
+  test('can handle a click event', async function(assert) {
+    assert.expect(2);
+
+    this.owner.register(
+      'component:x-foo',
+      Component.extend({
+        click() {
+          assert.ok(true, 'click was fired');
+        },
+      })
+    );
+    this.owner.register('template:components/x-foo', hbs`<button>Click me!</button>`);
+
+    await this.render(hbs`{{x-foo}}`);
+
+    assert.equal(this.element.textContent, 'Click me!', 'precond - component was rendered');
+    click(this.element.querySelector('button'));
+  });
+
+  test('can use action based event handling', async function(assert) {
+    assert.expect(2);
+
+    this.owner.register(
+      'component:x-foo',
+      Component.extend({
+        actions: {
+          clicked() {
+            assert.ok(true, 'click was fired');
+          },
+        },
+      })
+    );
+    this.owner.register(
+      'template:components/x-foo',
+      hbs`<button {{action 'clicked'}}>Click me!</button>`
+    );
+
+    await this.render(hbs`{{x-foo}}`);
+
+    assert.equal(this.element.textContent, 'Click me!', 'precond - component was rendered');
+    click(this.element.querySelector('button'));
```

Comment:
I assume at some point passing in just a CSS selector will also be supported?

Reply:
Ya, these event helpers are only internal test things for now. My next RFC will be for adding the real ones here (and they will essentially have the same API as ember-native-dom-helpers).

Reply:
would that mean deprecating `ember-native-dom-helpers`?

Reply:
I don’t know, but possibly. The main differences from what we need here is that we do not want to defer to Ember’s own internal application test globals. It’s definitely possible to do this transition a number of ways...